### PR TITLE
Add preliminary 24 player support

### PR DESCRIPTION
**Pull request by Hedede**
- Replaced hardcoded value with MAX_SLOTS
- Did some testing on bnet with a few 24 player maps

### CODE REVIEW

**Comment by JFGHT** [comment]
Hello,

why is this branch not merged?

File: src/bnetprotocol.cpp [hunk 1]
```diff
@@ -491,7 +491,7 @@ std::vector<uint8_t> CBNETProtocol::SEND_SID_STARTADVEX3(uint8_t state, const st
     AppendByteArray(packet, CustomGame, 4);                // Custom Game
     AppendByteArrayFast(packet, gameName);                 // Game Name
     packet.push_back(0);                                   // Game Password is NULL
-    packet.push_back(98);                                  // Slots Free (ascii 98 = char 'b' = 11 slots free) - note: do not reduce this as this is the # of PID's Warcraft III will allocate
+    packet.push_back(110);                                 // Slots Free (ascii 98 = char 'n' = 23 slots free) - note: do not reduce this as this is the # of PID's Warcraft III will allocate
```
Comment (Josko) [inline src/bnetprotocol.cpp]: Not trying to be nitpicky but could you please update the comment to say `ascii 110`?

**Comment by Josko** [comment]
> Hello,
> 
> why is this branch not merged?

Hello, mostly because this is not my job but I still have to review it.

**Comment by JFGHT** [comment]
I just thought you forgot about it, didn't want to annoy you.

Btw, is there any way I can contact you directly?

**Comment by Josko** [comment]
Thank you!

**Pull request merged**